framework/src/outputs/Checkpoint.C

```diff
@@ -183,8 +183,12 @@ Checkpoint::output()
 void
 Checkpoint::updateCheckpointFiles(CheckpointFileNames file_struct)
 {
-  // Update the list of stored files
-  _file_names.push_back(file_struct);
+  // Update the list of stored files if it does not already contain file_struct
+  // This ensures that we don't store duplicates, which will cause issues when
```

Comment:
> This ensures that we don't store duplicates

maybe adapt the language for clarity. there could be a duplicate in the vector, but when we re-store a checkpoint with the same name, it could be a different state still (in which case we do want to overwrite the previous file)